src/pbrt/cpu/integrators.cpp

```diff
@@ -2867,6 +2867,8 @@ void SPPMIntegrator::Render() {
             Options->disableWavelengthJitter ? Float(0.5) : RadicalInverse(1, iter);
         const SampledWavelengths passLambda = film.SampleWavelengths(uLambda);
 
+        Float timeSample = samplerPrototype.Get1D();
```

Comment:
I think this isn't quite right: you'll end up taking a sample from successive dimensions of the sampler, which doesn't necessarily give you well-distributed samples (and for some like Halton, will wrap around and start repeating values after the prime number table is used up.)

Note above that `RadicalInverse()` is used to generate a good set of samples for wavelength sampling. I think it would work well to instead do `timeSample = RadicalInverse(2, iter)` and just use the next Halton dimension for times. That way you get samples that are not correlated with the wavelength samples.

Reply:
I see. That makes sense, thank you for pointing that out!